### DISCUSSION TITLE
Account model will not be validated before creating

### DISCUSSION THREAD

**Discussion by FLX-0x00** [janko/rodauth-omniauth#15] [General]
If you have extended the `Account` model for example the `domain` with `belongs_to :domain, required: true` or using `before_validation` the creation of a new account through rodauth_omniauth after successfull callback phase ignores the Rails model validations and directly insert the invalid account. Of course there is a `before_omniauth_create_account` (currently I do not know how to redirect in this function if something is not valid) but maybe its more "rails" style to follow the model's validations. Or is there anything I completly overlooked?

Have a great day!

**Comment by janko** [comment]
You could run Active Record validations like this:

```rb
before_omniauth_create_account do
  if rails_account.invalid?
    validation_error = rails_account.errors.full_messages.first
    set_redirect_error_flash validation_error
    redirect omniauth_login_failure_redirect
  end
end
```

You might need to assign the domain to the model instance before running validations.

**Reply by FLX-0x00** [reply]
This looks very nice - thanks for taking the time to review this. We have some custom validations and do not want users (with or without sso) to be able to register with any given domain. We only allow domains that are in the database (whitelist). I have implemented another more specific solution, but your code looks more like a general solution. This is maybe a thing that can be implemented in the gem itself or at least get some audience in the readme file =)

**Reply by janko** [reply]
Good point, added a [wiki page](https://github.com/janko/rodauth-rails/wiki/Run-Model-Validations).
